security/src/main/java/io/micronaut/security/rules/InterceptUrlMapRule.java

```diff
@@ -83,12 +87,22 @@ public SecurityRuleResult check(HttpRequest request, @Nullable RouteMatch routeM
         final String path = request.getUri().getPath();
         final HttpMethod httpMethod = request.getMethod();
 
+        Predicate<InterceptUrlMapPattern> exactMatch = p -> pathMatcher.matches(p.getPattern(), path) && p.getHttpMethod().isPresent() && httpMethod.equals(p.getHttpMethod().get());
+        Predicate<InterceptUrlMapPattern> uriPatternMatch = p -> pathMatcher.matches(p.getPattern(), path) && p.getHttpMethod().map(method -> method.equals(httpMethod)).orElse(true);
```

Comment:
Shouldn't this just check that `!p.getHttpMethod().isPresent()`

Reply:
Good catch. Changed the predicate